src/proto/mod.rs

```diff
@@ -232,6 +233,10 @@ where
             self.outbox
                 .write_i32::<BigEndian>(xid)
                 .expect("Vec::write should never fail");
+            // opcode
+            self.outbox
+                .write_i32::<BigEndian>(item.opcode() as i32)
+                .expect("Vec::write should never fail");
```

Comment:
This doesn't seem right? Each request writer already writes out the opcode manually.

Reply:
Ah, I see you removed that. That won't quite work, because `Auth` for example specifically does _not_ write out the opcode!

Reply:
By `Auth` do you mean `Request::Connect`? Because that's already handled specially in the `if` branch of this conditional. I see there's a separate `OpCode::Auth` but that doesn't appear to be used anywhere.

Reply:
Ah, yes, I was thinking of `OpCode::Auth`, which has since been removed. Ignore me -- I was remembering an older version of the code :) Writing out the opcode here seems like a good change!